lib/fs/basicfs.go

```diff
@@ -70,6 +70,17 @@ func newBasicFilesystem(root string, opts ...Option) *BasicFilesystem {
 	sep := string(filepath.Separator)
 	root = filepath.Clean(filepath.Dir(root + sep))
 
+	if build.IsIOS && !filepath.IsAbs(root) && root[0] != '~' {
+		newroot, err2 := rooted(root, "~/Documents")
+		if err2 == nil {
+			root = newroot
+		} else {
+			l.Warnln("Illegal folder", root, "-", err2)
+			// Cannot error from here so use an unwritable path that will fail later
+			root = "~/bad"
+		}
+	}
+
```

Comment:
What's going on here and why is it required? Can we ensure that the root is set correctly by the caller instead?

Reply:
This was carried over from Mobius. It seems they put `path="someFolder"` in their config and want it to be dynamically resolved to the actual path at runtime. I followed the same approach but it's fine to remove it, let the caller deal with it.